File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Domain/Participation.java
package Concurs.Domain;

import java.util.UUID;

public class Participation extends Entity<UUID>{
    private Child child;
    private Event event;

    public Participation(UUID id, Child ch, Event ev) {
        this.setId(id);
        this.child = ch;
        this.event = ev;
    }

    public Participation(Child ch, Event ev) {
        this.setId(UUID.randomUUID());;
        this.child = ch;
        this.event = ev;
    }

    public Child getChild() {
        return child;
    }

    public void setChild(Child child) {
        this.child = child;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }
}

File: Children-Competition-C#/Domain/Participation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concurs.Domain
{
    internal class Participation : Entity<Guid>
    {
        public Child Child { get; set; }
        public Event Event { get; set; }

        public Participation(Child ch, Event ev)
        {
            this.Id = Guid.NewGuid();
            this.Child = ch;
            this.Event = ev;
        }
        public Participation(Guid guid, Child ch, Event ev)
        {
            this.Id = guid;
            this.Child = ch;
            this.Event = ev;
        }
    }
}
